File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.java
class Solution {
    public int maxVowels(String s, int k) {
        
        var vowels = Set.of('a', 'e', 'i', 'o', 'u');
        
        int res=0;
        int count=0;
        
        for(int i=0; i<s.length(); i++) {
            
            if(vowels.contains(s.charAt(i))) {
                count++;
            }
            
            if(i>=k && vowels.contains(s.charAt(i-k))) {
                
                count--;
                
            }
            
            res = Math.max(res, count);
        }
        
        return res;
        
    }
}

File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        
        sets = {'a', 'e', 'i', 'o', 'u'}
        
        res = count =0
        
        for i, c in enumerate(s):
            
            if c in sets:
                
                count+=1
            
            if i>=k and s[i-k] in sets:
                
                count-=1
            
            res = max(res, count)
            
        
        return res
                